### PR TITLE
feat: Submission grid server paging

### PR DESCRIPTION
**Pull request by orilux**
## Description

- Adds server-backed pagination for the submissions grid.
- Reads `page` and `pageSize` from the URL and sends them to the submissions API.
- Keeps pagination state in sync with the URL for shareable/bookmarkable grid views.
- Resets pagination to page 1 when filters change or reset.
- Caps URL-provided `pageSize` to the supported UI range.
- Redirects invalid/default/out-of-range paging URLs to canonical URLs.
- Updates the submissions table to support externally controlled pagination.
- Keeps generic empty-table fallback behavior for edge cases where a page has no rows.
- Adds/updates focused tests for submissions filters, empty states, and pagination URL updates.

## Type of Change
Check all that apply.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Dependency update
- [ ] Refactoring (non-breaking change which improves code quality)
- [ ] Other (please describe)

## Checklist
- My code follows the style guidelines of this project
- I have performed a self-review of my own code
- I have commented my code, particularly in hard-to-understand areas
- I have made corresponding changes to the documentation
- My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- New and existing tests pass locally with my changes
- Any dependent changes have been merged and published in downstream modules

> [!NOTE]
> - [x] I confirm that my Pull Request follows all the checklist requirements above

## Screenshots
If applicable, add screenshots to help explain your changes.

## Release Notes

* **New Features**
  * Added pagination support to submissions table with page navigation and customizable page size.
  * New empty-state messages: "No submissions yet" with share option and "No submissions match these filters" with reset action.
  * Disabled filter, export, and column-view controls appropriately during empty or filtered states.

* **Tests**
  * Added comprehensive test coverage for pagination behavior, empty states, and toolbar disabled states.

### CODE REVIEW

File: features/submissions/ui/table/table-pagination.tsx [hunk 1]
```diff
@@ -16,14 +16,15 @@ import {
 
 interface TablePagination<TData> {
   table: Table<TData>;
+  totalRows?: number;
 }
 
-export function TablePagination<TData>({ table }: TablePagination<TData>) {
+export function TablePagination<TData>({ table, totalRows }: TablePagination<TData>) {
```
[the remark this answers is not in the document]
Comment (orilux) [reply features/submissions/ui/table/table-pagination.tsx:22]: Addressed

File: app/(main)/forms/[formId]/submissions/ui/submissions-with-filters.test.tsx [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+      initialStatus: ["new"],
+      totalRecords: 2,
+      totalPages: 1,
+    });
+
+    // Assert
+    screen.getByText("No submissions match these filters");
+    expect(screen.queryByText("No submissions yet")).toBeNull();
+
+    fireEvent.click(screen.getAllByRole("button", { name: /reset filters/i })[1]);
```
[the remark this answers is not in the document]
Comment (orilux) [reply app/(main)/forms/[formId]/submissions/ui/submissions-with-filters.test.tsx]: Addressed

**Review by PeterMarinov — APPROVED** [review]
Approved
